src/utils/Logger.ts

```diff
@@ -0,0 +1,35 @@
+import { LogInterface } from "../interfaces/LogInterface";
+import MessageType from "../types/MessageType";
+
+export class Logger implements LogInterface {
+  public doActive: boolean;
+
+  debug(msg: string, supportingData?: any[]) {
+    this.emitLogMessage("debug", msg, supportingData);
+  }
+  info(msg: string, supportingData?: any[]) {
+    this.emitLogMessage("info", msg, supportingData);
+  }
+  warn(msg: string, supportingData?: any[]): void {
+    this.emitLogMessage("warn", msg, supportingData);
+  }
+  error(msg: string, supportingData?: any[]): void {
+    this.emitLogMessage("error", msg, supportingData);
+  }
+  log(msg: string, supportingData?: any[]): void {
+    this.emitLogMessage("log", msg, supportingData);
+  }
+  private emitLogMessage(
+    msgType: MessageType,
+    msg: string,
+    supportingData?: any[]
+  ) {
+    if (this.doActive) {
+      if (supportingData) console[msgType](msg, ...supportingData);
+      else console[msgType](msg);
+    } else
+      console[msg] = () => {
```

Comment:
I don't think we need this else logic at all as we just want logging  to be no-op if it's disabled